conbench/tests/populate_local_conbench.py

```diff
@@ -407,5 +412,143 @@ def create_benchmarks_with_history():
     print(benchmark_ids[0])
 
 
+def generate_synthetic_benchmark_history():
+
+    commits = [c.strip() for c in reversed(ARROW_COMMIT_HASH_LINES_50.splitlines())]
+    benchmark_name = "dummy-bench"
+
+    distr_mean = 20.0
+    # This is to simulate the real-world effect of there being a theoretical
+    # optimum: the fastest benchmark duration time with all sources of noise
+    # being silenced.
+    lower_bound = 17.5
+    slowdown_offset = distr_mean * 0.1
+    slowdown_lin = distr_mean * 0.1
+    distribution = statistics.NormalDist(mu=20.0, sigma=2)
```

Comment:
I really love using simulated data. And I don't think we are _just yet_ using the shape of this data to test our techniques, so this comment is slightly less relevant just now but will be soon.

Could we make this generator function generate some data and save that as a fixture (we could also set a seed, I believe which would effectively be the same thing, we would get the same "random" numbers each time)? Even with pretty liberal z-scores, in principle there is a (very small!) chance that an observation is outside our band of variability when generated this way. Of course, we could re-try those tests and we would see that it's back to normal, but depending on what we are changing, the failure might start a wild goose chase! With a fixture (or seed), the tests outcomes themselves are deterministic, but we can re-shuffle the data if we want to or need to easily cause we have the generator helper

Reply:
Great point with respect to reproducibility in a test suite. Will certainly be of relevance. I agree that something like a regression detection method is best tested against specific data sets. Testing against certain distributions the become re-sampled upon every test run sounds pretty advanced (but also cool).